src/cascadia/QueryExtension/AzureLLMProvider.cpp

```diff
@@ -0,0 +1,209 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+#include "pch.h"
+#include "AzureLLMProvider.h"
+#include "../../types/inc/utils.hpp"
+#include "LibraryResources.h"
+
+#include "AzureLLMProvider.g.cpp"
+#include "AzureResponse.g.cpp"
+
+using namespace winrt::Windows::Foundation;
+using namespace winrt::Windows::Foundation::Collections;
+using namespace winrt::Windows::UI::Core;
+using namespace winrt::Windows::UI::Xaml;
+using namespace winrt::Windows::UI::Xaml::Controls;
+using namespace winrt::Windows::System;
+namespace WWH = ::winrt::Windows::Web::Http;
+namespace WSS = ::winrt::Windows::Storage::Streams;
+namespace WDJ = ::winrt::Windows::Data::Json;
+
+static constexpr std::wstring_view acceptedModels[] = {
+    L"gpt-35-turbo",
+    L"gpt4",
+    L"gpt4-32k",
+    L"gpt4o",
+    L"gpt-35-turbo-16k"
+};
+static constexpr std::wstring_view acceptedSeverityLevel{ L"safe" };
+
+const std::wregex azureOpenAIEndpointRegex{ LR"(^https.*openai\.azure\.com)" };
```

Comment:
`.*` is technically not really what you want because this matches for instace
```
https://hecker.io/hijacked/openai.azure.com/harr/harr
^^^^^          .*          ^^^^^^^^^^^^^^^^
```

Instead, I'd recommend properly parsing the URI, getting the domain and then use `til::ends_with` from `til/string.h` to check what the suffix is.

Reply:
Added a domain check, unfortunately I don't think there's a "correct" suffix check since that one is going to contain the individual user's deployment name. For example, the URL looks something like this:

`https://<myAzureResourceName>.openai.azure.com/openai/deployments/<myDeploymentName>/chat/completions?api-version=2024-02-15-preview`

Reply:
I believe the problem you encountered is that you used the `Domain` property of the `Uri` class, but what you need to use is the `Host` property, as that's the technically correct term for the name. It's just that in our case we want to check whether the third-level domain name equals `.openai.azure.com`. But since not all URIs have a name, but only an IP address, etc., it's the "hostname" that we need to get & check.

That said, and unless I'm mistaken, your intention with the regex is still to test for a prefix and suffix in the URI. For instance, the parts you want to test in your example URI are:
```
 https://<myAzureResourceName>.openai .azure .com/openai/deployments/<myDeploymentName>/chat/completions?api-version=2024-02-15-preview
^https.*                       openai\.azure\.com
```

Right? That's effectively identical to the following pseudo-code:
```cpp
const auto scheme = parsedUri.SchemeName();
const auto host = parsedUri.Host();

if (scheme == "https" && til::ends_with(host, ".openai.azure.com"))
```

Reply:
This may be worth doing